queries/python.scm

```diff
@@ -0,0 +1,64 @@
+;; Generated by the following command:
+;; > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-python/d6210ceab11e8d812d4ab59c07c81458ec6e5184/src/node-types.json \
+;;   | jq '[.[] | select(.type == "_simple_statement" or .type == "_compound_statement") | .subtypes[].type]'
+[
+  (assert_statement)
+  (break_statement)
+  (class_definition)
+  (continue_statement)
+  (decorated_definition)
+  (delete_statement)
+  (exec_statement)
+  (expression_statement)
+  (for_statement)
+  (function_definition)
+  (future_import_statement)
+  (global_statement)
+  (if_statement)
+  (import_from_statement)
+  (import_statement)
+  (nonlocal_statement)
+  (pass_statement)
+  (print_statement)
+  (raise_statement)
+  (return_statement)
+  (try_statement)
+  (while_statement)
+  (with_statement)
+] @statement
+
+(
+  (function_definition
+    name: (_) @functionName
+    body: (_) @namedFunction.interior
+  ) @namedFunction @functionName.domain
+  (#set! onlyIfNotDescendantOfType decorated_definition)
+)
+(decorated_definition
+  (function_definition
+    name: (_) @functionName
+    body: (_) @namedFunction.interior
+  )
+) @namedFunction @functionName.domain
+
+(
+  (class_definition
+    name: (_) @className
+    body: (_) @class.interior
+  ) @class @className.domain
+  (#set! onlyIfNotDescendantOfType decorated_definition)
+)
+(decorated_definition
+  (class_definition
+    name: (_) @className
+    body: (_) @class.interior
+  )
+) @class @className.domain
```

Comment:
Note that this matcher and the above are the really awkward way that we have to implement the `?` construct from our own pattern language.  It's a bummer that it's so much more cumbersome in the native tree-sitter query language, but unfortunately we're kind of stuck with it.  If the language continues to feel too low-level, we can think about generating queries based on a higher level specification language 🤔